### ISSUE TITLE
RuntimeError: NetCDF: HDF error

### ISSUE THREAD

**Issue by amruthkiran94** [opendatacube/datacube-core#801]
Hello all,
I've followed the installation as per the documentation at `datacube-ui/docs` for both the datacube as well the datacube-ui installation. There a few problems with both of them. 

1. Installation for the datacube-core via `pip` and `virtualenv` or ` datacube_env` prompts with the following error while ingestion :
```
RuntimeError: NetCDF: HDF error
2019-09-26 16:17:11,021 9512 datacube-ingest ERROR Failed to create storage unit file (Exception: NetCDF: HDF error) 
Traceback (most recent call last):
  File "/home/localuser/Datacube/datacube-core/datacube/scripts/ingest.py", line 333, in process_tasks
    executor.result(future)
  File "/home/localuser/Datacube/datacube-core/datacube/executor.py", line 63, in result
    return func(*args, **kwargs)
  File "/home/localuser/Datacube/datacube-core/datacube/executor.py", line 33, in reraise
    raise t.with_traceback(e, traceback)
  File "/home/localuser/Datacube/datacube-core/datacube/executor.py", line 36, in get_ready
    result = SerialExecutor.result(futures[0])
  File "/home/localuser/Datacube/datacube-core/datacube/executor.py", line 63, in result
    return func(*args, **kwargs)
  File "/home/localuser/Datacube/datacube-core/datacube/scripts/ingest.py", line 279, in ingest_work
    storage_config=config['storage'])
  File "/home/localuser/Datacube/datacube-core/datacube/drivers/netcdf/driver.py", line 53, in write_dataset_to_storage
    **kwargs)
  File "/home/localuser/Datacube/datacube-core/datacube/drivers/netcdf/_write.py", line 91, in write_dataset_to_netcdf
    netcdfparams)
  File "/home/localuser/Datacube/datacube-core/datacube/drivers/netcdf/_write.py", line 41, in create_netcdf_storage_unit
    netcdf_writer.create_coordinate(nco, name, coord.values, coord.units)
  File "/home/localuser/Datacube/datacube-core/datacube/drivers/netcdf/writer.py", line 95, in create_coordinate
    var[:] = labels
  File "/home/localuser/Datacube/datacube-core/datacube/drivers/netcdf/_safestrings.py", line 42, in __setitem__
    self._wrapped.__setitem__(key, value)
  File "netCDF4/_netCDF4.pyx", line 4950, in netCDF4._netCDF4.Variable.__setitem__
  File "netCDF4/_netCDF4.pyx", line 5229, in netCDF4._netCDF4.Variable._put
  File "netCDF4/_netCDF4.pyx", line 1887, in netCDF4._netCDF4._ensure_nc_success
RuntimeError: NetCDF: HDF error
0 successful, 0 failed
```
Running on Ubuntu 18.04 LTS, Datacube Latest pull from git. GDAL = 2.4.2

Disregarding this, i went ahead with the datacube-ui installation and it went as expected without any errors. I was able to get to the UI homepage. Although I couldn't view my dataset as it wasn't ingested properly.

2. Reinstalling this time via `conda` and `cubeenv` as per the official documentation, the datacube-core installs perfectly as expected and ingestion also runs fine. The problem arises when coupling with datacube-ui now. 
```
Target WSGI script '/home/localuser/Datacube/data_cube_ui/data_cube_ui/wsgi.py' cannot be loaded as Python module.
```
```
ValueError: failed to parse CPython sys.version: '3.6.8
```

TL;DR- 
Installing via `pip` datacube-core doesn't work but datacube-ui does. Installing via `conda` datacube-ui works but datacube-core doesn't.

I would like to quote Mr. John Rattz from a slack thread-

>  Are you using the cubeenv Conda environment or the datacube_env virtualenv environment? You can't use both. Your safest bet is to follow this document to install the ODC: https://github.com/ceos-seo/data_cube_ui/blob/master/docs/open_data_cube_install.md and this document to install the UI: https://github.com/ceos-seo/data_cube_ui/blob/master/docs/ui_install.md
If you want to use cubeenv, don't install the datacube_env environment or source it. If you want the UI installation to be simpler, don't use cubeenv and instead use virtualenv as it describes. We plan to update our documentation to use Conda instead of virtualenv soon.

Any suggestions are welcome!

Thanks!

**Comment by benjimin** [comment]
Regards part 2 (datacube installed per the recommendations in the core docs; datacube apparently working but datacube-ui broken) perhaps should raise issue to data_cube_ui directly?

**Comment by woodcockr** [comment]
No activity on issue - closing
